src/kangaroo.rs

```diff
@@ -0,0 +1,192 @@
+use crunchy::unroll;
+use super::{PLEN, KeccakFamily, RHO, PI, Permutation};
+
+const RC: [u64; 12] = [
+	0x000000008000808b,
+	0x800000000000008b,
+	0x8000000000008089,
+	0x8000000000008003,
+	0x8000000000008002,
+	0x8000000000000080,
+	0x000000000000800a,
+	0x800000008000000a,
+	0x8000000080008081,
+	0x8000000000008080,
+	0x0000000080000001,
+	0x8000000080008008,
+];
+
+#[allow(unused_assignments)]
+pub fn keccakf(a: &mut [u64; PLEN]) {
+    for i in 0..12 {
```

Comment:
why not unroll this, does the function become too big for caches?

Reply:
Can you extract `12 ` to `ROUNDS` constant?

Reply:
> why not unroll this, does the function become too big for caches?

We've benchmarked this and unrolled version of this loop was actually slower